language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/util/rename/RenameUtil.java

```diff
@@ -89,7 +90,8 @@ private RenameUtil() {
                 })
                 .orElse(null);
         // Check if token at cursor is an identifier
-        if (!(tokenAtCursor instanceof IdentifierToken) || CommonUtil.isKeyword(tokenAtCursor.text())) {
+        if (!(tokenAtCursor instanceof IdentifierToken) || CommonUtil.isKeyword(tokenAtCursor.text()) 
+                || CommonUtil.SELF_KW.equals(tokenAtCursor.text())) {
```

Comment:
As per the language semantics, `self` is not always a keyword. Self is a keyword only within the object constructor expression, and classes. Hence we will need to address that particular scoping check here as well.

Reply:
+1. We can lookup the parents of the `tokenAtCursor` and check if there's a class/object as a parent. Probably move that logic to a separate method.